src/core/tranche/distributor.sol

```diff
@@ -0,0 +1,198 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+
+contract PileLike {
+    function want() public returns (int);
+}
+
+contract QuantLike {
+    uint public debt;
+}
+
+contract OperatorLike {
+
+    function borrow(address, uint) public;
+    function repay(address, uint) public;
+    function balance() public returns (uint);
+    function file(bytes32, bool) public;
+
+    bool public supplyActive;
+    bool public redeemActive;
+
+    QuantLike public quant;
+}
+
+contract Distributor is DSNote {
+
+    // --- Tranches ---
+
+    struct Tranche {
+        bool senior;
```

Comment:
If we are doing that we can't make it generic. 

We should either make it generic or only work for two tranches